R/unitConverter.R

```diff
@@ -0,0 +1,103 @@
+#' Convert to common units
+#'
+#' @description
+#'
+#' When multiple (observed and/or simulated) datasets are present in a data
+#' frame, they are likely to have different units. This function helps to
+#' convert them to a common unit specified by the user.
+#'
+#' This is especially helpful while plotting since the quantities from different
+#' datasets to be plotted on the x-and y-axis need to have same units to be
+#' meaningfully compared.
+#'
+#' @return A data frame with measurement columns transformed to have common units.
+#'
+#' @param data A data frame (or a tibble).
+#' @param xUnit,yUnit Target units for `xValues` and `yValues`, respectively. If
+#'   not specified (`NULL`), first of the existing units in the respective
+#'   columns (`xUnit` and `yUnit`) will be selected as the common unit. For
+#'   available dimensions and units, see `ospsuite::ospDimensions` and
+#'   `ospsuite::ospUnits`, respectively.
+#'
+#' @seealso toUnit
+#'
+#' @examples
+#'
+#' # small dataframe to illustrate the conversion
+#' (df <- dplyr::tibble(
+#'   xValues = c(15, 30, 60), xUnit = "min", xDimension = "Time",
+#'   yValues = c(0.25, 45, 78), yUnit = c("", "%", "%"), yDimension = c("Fraction", "Fraction", "Fraction"),
+#'   molWeight = c(12.5, 10, 5)
+#' ))
+#'
+#' ospsuite:::.unitConverter(df)
+#' ospsuite:::.unitConverter(df, xUnit = ospUnits$Time$h)
+#' ospsuite:::.unitConverter(df, yUnit = ospUnits$Fraction$`%`)
+#' ospsuite:::.unitConverter(df, xUnit = ospUnits$Time$h, yUnit = ospUnits$Fraction$`%`)
+#'
+#' @keywords internal
```

Comment:
This is an internal function, but never hurts to have it documented. No `.Rd` files will be created for it.

Reply:
Documentation is always better than no documentation! :D